server/router/rider.ts

```diff
@@ -60,6 +62,24 @@ router.get('/:id/favorites', validateUser('User'), (req, res) => {
   });
 });
 
+// Get current/soonest ride (within next 30 min) of rider, if exists
+router.get('/:id/currentride', validateUser('Rider'), (req, res) => {
+  const { params: { id } } = req;
+  db.getById(res, Rider, id, tableName, () => {
+    const now = moment().toISOString();
+    const end = moment().add(30, 'minutes').toISOString();
```

Comment:
Also, the "end" const might get confused with the "endTime" ride field, so renaming it to "windowEnd" or something might help avoid confusion